pkgs/applications/emulators/emulationstation/default.nix

```diff
@@ -1,34 +1,72 @@
-{ lib, stdenv, fetchFromGitHub, pkg-config, cmake, curl, boost, eigen
-, freeimage, freetype, libGLU, libGL, SDL2, alsa-lib, libarchive
-, fetchpatch }:
+{ lib
+, SDL2
+, alsa-lib
+, boost
+, cmake
+, curl
+, eigen
+, fetchFromGitHub
+, fetchpatch
+, freeimage
+, freetype
+, libGL
+, libGLU
+, libarchive
+, pkg-config
+, stdenv
+, vlc
+, rapidjson
+, pugixml
+}:
 
 stdenv.mkDerivation {
   pname = "emulationstation";
-  version = "2.0.1a";
+  version = "2.12.0rp-dev";
 
   src = fetchFromGitHub {
-    owner = "Aloshi";
+    owner = "RetroPie";
     repo = "EmulationStation";
-    rev = "646bede3d9ec0acf0ae378415edac136774a66c5";
-    sha256 = "0cm0sq2wri2l9cvab1l0g02za59q7klj0h3p028vr96n6njj4w9v";
+    rev = "30cbdeb7cd667ac9d6a836995a17004e8d84339a";
+    sha256 = "sha256-jA3wLf/s9UlsP8h66aMOLZmLfTu7Zjhmt+BX6lmBreM=";
   };
 
-  patches = [
-    (fetchpatch {
-      url = "https://github.com/Aloshi/EmulationStation/commit/49ccd8fc7a7b1dfd974fc57eb13317c42842f22c.patch";
-      sha256 = "1v5d81l7bav0k5z4vybrc3rjcysph6lkm5pcfr6m42wlz7jmjw0p";
-    })
-  ];
+  patchPhase = ''
+    substituteInPlace */src/*.cpp */src/*/*.cpp */src/*.h \
```

Comment:
This should be upstreamed.

Reply:
They build `pugixml` themselves in a subdirectory themselves instead of using a package, arguing in the docs that this lib is not usually available as a package on the different distros.

I can try upstream that, but i think this will take some more time because that kicks off another situation where people have been doing things like this for years and then a nixer comes out from nothing and nixplains that this is wrong.

Reply:
They basically "[vendored](https://stackoverflow.com/questions/26217488/what-is-vendoring)" pugixml.
In this case the term would be "gentoosplaining":

https://blogs.gentoo.org/mgorny/2021/02/19/the-modern-packagers-security-nightmare/

https://wiki.gentoo.org/wiki/Why_not_bundle_dependencies

I am a bit divided on this issue.
Usually I would follow upstream and use their vendored code instead of import our own.

Regardless of personal opinions whether vendoring is good or bad, the package developers should be able to use the system-wide pugixml as an alternative, without recurring to such a heavy patching.

About this, have you tried to build this with their vendored code?

Reply:
I upstreamed this change: https://github.com/RetroPie/EmulationStation/pull/821 

Shall this PR stay open until that is merged?

Reply:
Huh, that's fine! 
I don't believe they will generate a new release.
If the patch is working and you believe the upstream will take too much time to merge it, then you can for now download the file generated by https://github.com/RetroPie/EmulationStation/pull/821.patch, rename it and include as a patch:

```nix
patches = [
    # TODO: remove when merged upstream 
    # (https://github.com/RetroPie/EmulationStation/pull/821)
   ./fix-pugixml-detection.patch
];
```

Reply:
Good idea, done!